### DISCUSSION TITLE
Need to test video player package in mobile

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#10] [Installation] [answered]
Hello Swayanjit Sir,

We need your help to test the latest player changes with the mobile app. 
Can you please help 

@coolbung  @baljeetsingh  @swayangjit

**Comment by sharathkashyap** [comment, marked as answer]
@Xitija In package.json in Mobile App project. Kindly change the version of @project-sunbird/sunbird-video-player to latest version available in npm.
[package.json](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/blob/release-4.9.5/package.json)
`"@project-sunbird/sunbird-video-player-v9": "4.7.1",`

Note: There is a seperate process if you want to install it locally.

cc: @swayangjit

**Reply by Xitija** [reply]
Yes it is a local package , I need to test it in the app

**Reply by sharathkashyap** [reply]
@Xitija Then the process is as follows. 
1st Terminal
a) Navigate to sunbird-video-player project
b) Build the repo
c) Verify if the dist got generated.

2nd Terminal
a) Navigate to Mobile Project
b) npm i ../@relative-dir-location/sunbird-video-player/dist

This should deploy the local changes onto Mobile Project
c) Build Mobile

**Reply by Xitija** [reply]
In the mobile project where exactly the package has to be added at what path

**Reply by sharathkashyap** [reply]
Navigate to Mobile Project root
and run 
`npm i ../@relative-dir-location-of-sunbird-video-player/sunbird-video-player/dist`
Adding packages to node_modules will be handled by previous step.

There is no need for a separate addition of these packages.

You can verify the packages inside node_modules folder.

cc: @luckynvdu

**Reply by Xitija** [reply]
Hello @swayangjit ,

I am trying to run an Interactive Video in the mobile app .
When the video reaches the Question Set the UI is breaking .

![image](https://user-images.githubusercontent.com/65657373/170449190-49ffdc99-e2bf-4913-b450-d20c4a662495.png)


We want to open the Question Set in landscape mode ,
Can you please tell what code is to be added / updated for doing that ?

**Reply by swayangjit** [reply]
@Xitija  Please share the ecar file  so that I can check and let you know the solution.

**Reply by Xitija** [reply]
We are just checking online content for now and not using ecar as ecar is not getting imported properly , do_id of the content shown in the snapshot is do_21310353608830976014671

**Reply by swayangjit** [reply]
@Xitija This is in which environment ?

**Reply by Xitija** [reply]
Sorry I didnt get you .This is app is built on local setup after addig the video player package , the UI breaking issue persists in the existing app , also it is pointing to https://staging.sunbirded.org/  .

**Reply by swayangjit** [reply]
@Xitija  Ok it's in staging. I will check in staging and get back to you

**Reply by Xitija** [reply]
I tried importing ecar in the mobile but it is failing.
![image](https://user-images.githubusercontent.com/65657373/170643916-cb0a785f-a6cb-4a29-9bca-a426e94adeaa.png)
How do we get to know the failure reason ?
Any logs that get generated ?
Or is there any way we can turn on the logs for us to understand ?
Can we make use of Android Studio for getting more information?

So that would help us fix ecar imports , can you please guide @swayangjit

**Reply by sharathkashyap** [reply]
@Xitija 
`How do we get to know the failure reason ?`
You may need to install mobile-sdk as local node package and debug the reasons for failure.
`"@project-sunbird/sunbird-sdk": "4.9.0",`

`Any logs that get generated ?`
Error Logs do get generated in Chrome Console On Each Step of Import.
https://github.com/Sunbird-Ed/sunbird-mobile-sdk/blob/release-4.9.0/src/content/impl/content-service-impl.ts
Method Name : importEcar
line: 390

Note: Kindly request you to open a seperate thread for import ecar issue. Will be locking up this conversation as the initial issue seems resolved. 
cc: @swayangjit @luckynvdu
